R/sdf_utils.R

```diff
@@ -0,0 +1,32 @@
+#' Fast cbind for Spark DataFrames
+#'
+#' This is a version of `sdf_bind_cols` that works by zipping
+#' RDDs. From the API docs: "Assumes that the two RDDs have the
+#' *same number of partitions* and the *same number of elements
+#' in each partition* (e.g. one was made through a map on the
+#' other)."
+#'
+#' @keywords internal
+#'
+#' @param ... Spark DataFrames to cbind
+sdf_fast_bind_cols <- function(...) {
```

Comment:
I think that sounds good. Alternatively, we could hide the validation code behind some R / sparklyr configuration option?

Either way, maybe let's leave that for a separate PR in case we find that we want to use `sdf_fast_bind_cols()` in other contexts where it's less obvious that the required preconditions are met (for some reason).